rskj-core/src/main/java/org/ethereum/util/BIUtil.java

```diff
@@ -130,4 +130,13 @@ public static boolean isIn20PercentRange(BigInteger first, BigInteger second) {
     public static BigInteger max(BigInteger first, BigInteger second) {
         return first.compareTo(second) < 0 ? second : first;
     }
+    /**
+    * Returns a result of safe addition of two {@code int} values
+    * {@code Integer.MAX_VALUE} is returned if overflow occurs
+    */
+    public static int addSafely(int a, int b) {
```

Comment:
Be careful as this only works for positive numbers

Reply:
The only usage of this is with lengths ( always positive )

Reply:
This is definitely in the wrong place. It isn't really a `BigInteger` util, so it should be moved somewhere else.

I would also argue this isn't _safe in general_, but just _safe in this case_. Slapping _safe_ to a method name might make people too confident.

Reply:
OK, we need more context here. This change is based on https://github.com/ethereum/ethereumj/commit/aac4020b9af52af65f0e354f03c6a6970426c0fe#diff-add6709d11898499ee331496844ebc0aR138

So I must say I don't like what they're doing and that I don't think that we should be copying them in this case. Can we at least document why we're using `MAX_VALUE` instead of throwing an exception?? I would look into other implementations of EIP-198.

If this value never really gets used because of other related statements (e.g. I see there's a `Math.min` being used), then don't create this function at all.

I'll test locally and let you know.